### ISSUE TITLE
[doc] 9.2  Enabling Eirini

### ISSUE THREAD

**Issue by it-slav** [SUSE/doc-cap#1002]
9.2  Enabling Eirini:

https://documentation.suse.com/suse-cap/2.1/html/cap-guides/cha-cap-depl-eirini.html#sec-cap-eirini-enable

I do not understand bullet point 3 and it does not work:
> helm3 install metrics-server stable/metrics-server --set args[0]="--kubelet-preferred-address-types=InternalIP" --set args[1]="--kubelet-insecure-tls"
Error: failed to download "stable/metrics-server" (hint: running `helm repo update` may help)

**Comment by btat** [comment]
```
 In order for Eirini to report application metrics, Metrics Server (link xlink:href="https://github.com/kubernetes-sigs/metrics-server"/> must be installed.

Note that --kubelet-insecure-tls is not recommended for production usage, but can be useful in test clusters with self-signed Kubelet serving certificates. For production, use --tls-private-key-file. 
```

Step 3 has the two above sentences, could you clarify which part you're not understanding. Is is why the step is needed or the part about the different parameters for the command? 

As listed in 9.1, `Eirini requires the k8s-metrics-server to be installed on the Kubernetes environment where SUSE Cloud Application Platform is installed in order for Stratos Metrics to work. `, this is just the step for you to install the Kubernetes Metrics Server

For the command, have you run `helm repo update` as suggested? `stable` is a default Helm repository.

**Comment by zoopster** [comment]
@it-slav where does your stable repository point?

**Comment by it-slav** [comment]
> helm3 repo ls
NAME       	URL                                             
suse       	https://kubernetes-charts.suse.com/             
helm-stable	https://kubernetes-charts.storage.googleapis.com
stable     	https://charts.helm.sh/stable

**Comment by it-slav** [comment]
But now the error is different.
```
helm3 install metrics-server stable/metrics-server --set args[0]="--kubelet-preferred-address-types=InternalIP" --set args[1]="--kubelet-insecure-tls"
Error: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "system:metrics-server" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "metrics-server"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "default"
```

**Comment by it-slav** [comment]
@btat both sentences are vague. 

> In order for Eirini to report application metrics, Metrics Server (link xlink:href="https://github.com/kubernetes-sigs/metrics-server"/> must be installed.
Is application metrics necessary for Stratos to get app status? For me they are grey.
I did install metrics following other parts in the CAP documentation (https://documentation.suse.com/suse-cap/2.1/single-html/cap-guides/#sec-cap-stratos-metrics), are there several? Do I need both? 

> Note that --kubelet-insecure-tls is not recommended for production usage, but can be useful in test clusters with self-signed Kubelet serving certificates. For production, use --tls-private-key-file.
I use ingress and lestencrypt certs, what do I need todo to get that working or is it irrelevant?

Where is that suggested that I need to run helm repo update?

**Comment by btat** [comment]
> But now the error is different.
> 
> ```
> helm3 install metrics-server stable/metrics-server --set args[0]="--kubelet-preferred-address-types=InternalIP" --set args[1]="--kubelet-insecure-tls"
> Error: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "system:metrics-server" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "metrics-server"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "default"
> ```

@it-slav can you run `kubectl get pods --all-namespaces | grep metrics-server` and see if there's already a `metrics-server` running

**Comment by btat** [comment]
> @btat both sentences are vague.
> 
> > In order for Eirini to report application metrics, Metrics Server (link xlink:href="[https://github.com/kubernetes-sigs/metrics-server"/>](https://github.com/kubernetes-sigs/metrics-server%22/%3E) must be installed.

With some `cf` cli commands (e.g. http://cli.cloudfoundry.org/en-US/v6/app.html) you can display things like cpu, memory, and disk usage of an app. These are the "application metrics" being referred to. When using Eirini, there may be issues displaying the correct state of these. Installing the Kubernetes Metrics Server ensures these are correctly reported.

> 
> Is application metrics necessary for Stratos to get app status? For me they are grey.
> I did install metrics following other parts in the CAP documentation (https://documentation.suse.com/suse-cap/2.1/single-html/cap-guides/#sec-cap-stratos-metrics), are there several? Do I need both?

Kubernetes Metrics Server and Stratos Metrics are different things. If you install Stratos Metrics, you need the Kubernetes Metrics Server in order for Stratos Metrics to work. This is noted in the release notes and earlier in the chapter (section 9.1).

> 
> > Note that --kubelet-insecure-tls is not recommended for production usage, but can be useful in test clusters with self-signed Kubelet serving certificates. For production, use --tls-private-key-file.
> 
> I use ingress and lestencrypt certs, what do I need todo to get that working or is it irrelevant?

I'm not too familiar with the different combinations, @satadruroy can you please provide guidance for this question (and provide additional background/info for my above responses, if needed).

> 
> Yes I did run helm3 repo update

**Comment by satadruroy** [comment]
I will go over @it-slav setup tomorrow and update.

**Comment by it-slav** [comment]
> ```
>  In order for Eirini to report application metrics, Metrics Server (link xlink:href="https://github.com/kubernetes-sigs/metrics-server"/> must be installed.
> 
> Note that --kubelet-insecure-tls is not recommended for production usage, but can be useful in test clusters with self-signed Kubelet serving certificates. For production, use --tls-private-key-file. 
> ```
> 
> Step 3 has the two above sentences, could you clarify which part you're not understanding. Is it why the step is needed or the part about the different parameters for the command?
> 
> As listed in 9.1, `Eirini requires the k8s-metrics-server to be installed on the Kubernetes environment where SUSE Cloud Application Platform is installed in order for Stratos Metrics to work `, this is just the step for you to install the Kubernetes Metrics Server

This is a problem with the documentation. The information is spread out all over. It is very hard to figure out which parts are relevant for my setup and how changing one thing affect another part.
> 
> For the command, have you run `helm repo update` as suggested? `stable` is a default Helm repository.

**Comment by it-slav** [comment]
> > But now the error is different.
> > ```
> > helm3 install metrics-server stable/metrics-server --set args[0]="--kubelet-preferred-address-types=InternalIP" --set args[1]="--kubelet-insecure-tls"
> > Error: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "system:metrics-server" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "metrics-server"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "default"
> > ```
> 
> @it-slav can you run `kubectl get pods --all-namespaces | grep metrics-server` and see if there's already a `metrics-server` running

```> kubectl get pods --all-namespaces | grep metrics-server
kube-system              metrics-server-bf97646d6-6khqt                       1/1     Running            0          7d
kube-system              metrics-server-bf97646d6-rbfds                       1/1     Running            0          7d
``` 
Yes, I have metrics running, following https://github.com/SUSE/stratos-metrics. That documentation is also very hard to follow and understand.
If I have that, is the ```helm3 install metrics-server stable/metrics-server --set args[0]="--kubelet-preferred-address-types=InternalIP" --set args[1]="--kubelet-insecure-tls"``` not needed? How shall I know that?

After reading crossreferenced documents back and forth with myriads of parameters that are vaguely documented, it is very unlikely to get a correct configuration.
I might have the wrong expectation of the documentation?

**Comment by it-slav** [comment]
After other discussions I have learned that there is different metrics. One is Stratos and another is k8s metrics. Another thing that doc does not explain (it might be out of scope as that is expected knowledge).

I have followed the instructions at https://www.suse.com/c/how-to-use-lets-encrypt-certificates-with-ingress-controller-for-suse-caasp-and-suse-cap/ which include monitoring the local caasp cluster. If I am correct that has already installed the metrics-server.

Is that enough to get the eirini data?

**Comment by zoopster** [comment]
I was trying to verify that, but my caasp cluster was down. CaaS Platform installs the k8s metrics-server by default during installation.

**Comment by it-slav** [comment]
I have just installed caasp 4.5 on another system and there are metrics-server pods so I guess that is installed by default.
Is that enough to get the eirini data?

**Comment by btat** [comment]
For history: it-slav had a debugging session with Satadru regarding this.

Replaced by https://github.com/SUSE/doc-cap/issues/1007
